analyzer/src/traversal/expressions.rs

```diff
@@ -661,8 +661,18 @@ fn expr_call_value_attribute(
     if let fe::Expr::Attribute { value, attr } = &func.kind {
         let value_attributes = expr(Rc::clone(&scope), Rc::clone(&context), &value)?;
 
-        if let Type::Contract(contract) = value_attributes.typ {
-            return expr_call_contract_attribute(scope, context, contract, &attr.kind, args);
+        if let Type::Contract(contract) = &value_attributes.typ {
+            // We must ensure the expression is loaded onto the stack.
+            context
+                .borrow_mut()
+                .update_expression(value, value_attributes.clone().into_loaded()?);
```

Comment:
Originally, calls to contract fields would fail (e.g. `self.foo.get_my_string()`). This is because the call was being made to a non-existent contract at the address of the last 20 bytes of the storage pointer.